### ISSUE TITLE
Error When Fetching Nonce

### ISSUE THREAD

**Issue by alexkendall** [arpansharma7474/rn-braintree#2]
I was able to install the library and initialization works. After initializing the library in react-native the message `Success in setup Braintree Initialized Successfully`. Despite this, when attempting to fetch the card nonce, I am met with the error `Error in onCardPress Error: The operation couldn’t be completed. (com.braintreepayments.BTHTTPErrorDomain error 2.)
    at Object.promiseMethodWrapper [as getCardNonce]`. Any idea why this error is being caused, and if there's an issue in my code setup, or the sandbox setup?

**Comment by arpansharma7474** [comment]
Can you please share the code snippet?

**Comment by alexkendall** [comment]
@arpansharma7474 Can do.

```  
  useEffect(() => {
    RnBraintree.setup('sandbox_q4gs98q7_fw3wm932cq9vqq7m')
      .then(res => {
        console.log('Success in setup', res)
        RnBraintree.getCardNonce({
          name: 'Test User',
          number: '4111111111111111',
          expirationDate: '10/2024',
          cvv: '400'
        })
          .then(res => {
            console.log('Success getting nonce', res)
          })
          .catch(err => {
            console.log('Error in getting nonce', err)
          })
      })
      .catch(err => {
        console.log('Error in setup', err)
      })
  }, [])

**Comment by arpansharma7474** [comment]
I think the problem is that you are not calling getCardNonce after the setup is complete. Please move the getCardNonce call to inside of then block of setup. It should work.

**Comment by alexkendall** [comment]
So the setup was completing on time and I even refactored the code to the edited version above. I actually found out the issue and it was in RnBraintree.swift. Disabling [line 41](https://github.com/arpansharma7474/rn-braintree/blob/main/ios/RnBraintree.swift#L41) `card.shouldValidate = true` fixes the issue, and disabling this line in the Android java file fixes the issue for Android.

**Comment by arpansharma7474** [comment]
@alexkendall 
I have released a new version of the package(1.0.1), now you can pass shouldValidateCard as a boolean param to the getCardNonce() function.

Hope this helps you.

Happy Coding
